Duplicati/Library/Backend/AzureBlob/AzureBlobWrapper.cs

```diff
@@ -62,11 +62,18 @@ public string[] DnsNames
 
         public AzureBlobWrapper(string accountName, string accessKey, string containerName)
         {
-            _containerName = containerName;
+            OperationContext.GlobalSendingRequest += (sender, args) =>
+            {
+                string userAgent = "APN/1.0 Duplicati/2.0 AzureBlob/2.0 " + Microsoft.WindowsAzure.Storage.Shared.Protocol.Constants.HeaderConstants.UserAgent;

```

Comment:
I wouldn't put in a hard coded string like this, rather a constant. You could add a property to the class for setting this. Also if we want to supply a user agent to services (is this the only place we do it), so that services can see what program is being used (whether this is a good thing I will leave out of the discussion), it could be a global setting for consistency.

Reply:
In other places I have used the library's version:
https://github.com/duplicati/duplicati/blob/master/Duplicati/Library/Backend/WEBDAV/WEBDAV.cs#L371

Reply:
Sounds good to me. 👍

Reply:
> Cons: The service could target our users? I don't know, either way, with a opt-out option this would be mitigated.

Should that happen, we can discuss the appropriate action. Generally, a service acting on the User-Agent header is asking for trouble, but it is a nice option for service debugging, stats and customer support (i.e. "the software you are using has a know issue, please upgrade").

If a provider wants to block Duplicati requests, I don't think we should play cat-and-mouse with them. Their users should complain or change providers.

Reply:
@mikaelmello  I can make the change if you prefer?

Reply:
I guess that's better